server/routes/induction/update/previousWorkExperienceTypesUpdateController.test.ts

```diff
@@ -0,0 +1,310 @@
+      )
+
+      // Then
+      expect(res.redirect).toHaveBeenCalledWith('/plan/A1234BC/view/work-and-interests')
+      expect(mockedCreateOrUpdateInductionDtoMapper).not.toHaveBeenCalled()
+      expect(inductionService.updateInduction).not.toHaveBeenCalled()
+      expect(req.session.previousWorkExperienceTypesForm).toBeUndefined()
+      expect(req.session.inductionDto).toBeUndefined()
+    })
+
+    it('should update Induction given form is submitted where the only change is a removal of a work type other than OTHER', async () => {
+      // Given
+      req.user.token = 'some-token'
+      const prisonNumber = 'A1234BC'
+      req.params.prisonNumber = prisonNumber
+
+      const prisonerSummary = aValidPrisonerSummary()
+      req.session.prisonerSummary = prisonerSummary
+      const inductionDto = aLongQuestionSetInductionDto({ hasWorkedBefore: true })
+      // The induction has work experience of CONSTRUCTION and OTHER (Retail delivery)
+      req.session.inductionDto = inductionDto
+
+      const previousWorkExperienceTypesForm = {
+        typeOfWorkExperience: ['OTHER'],
+        typeOfWorkExperienceOther: 'Retail delivery',
+      }
+      req.body = previousWorkExperienceTypesForm
+      req.session.previousWorkExperienceTypesForm = undefined
+
+      const updateInductionDto = aLongQuestionSetUpdateInductionDto({ hasWorkedBefore: true })
+      mockedCreateOrUpdateInductionDtoMapper.mockReturnValueOnce(updateInductionDto)
+
+      mockedFormValidator.mockReturnValue(errors)
+      const expectedPreviousWorkExperiences: Array<PreviousWorkExperienceDto> = [
+        {
+          details: 'Self employed franchise operator delivering milk and associated diary products.',
+          experienceType: 'OTHER',
+          experienceTypeOther: 'Retail delivery',
+          role: 'Milkman',
+        },
+      ]
+
+      // When
+      await controller.submitPreviousWorkExperienceTypesForm(
+        req as undefined as Request,
+        res as undefined as Response,
+        next as undefined as NextFunction,
+      )
+
+      // Then
+      // Extract the first call to the mock and the second argument (i.e. the updated Induction)
+      const updatedInduction = mockedCreateOrUpdateInductionDtoMapper.mock.calls[0][1]
+      expect(mockedCreateOrUpdateInductionDtoMapper).toHaveBeenCalledWith(prisonerSummary.prisonId, updatedInduction)
+      expect(updatedInduction.previousWorkExperiences.experiences).toEqual(expectedPreviousWorkExperiences)
+
+      expect(inductionService.updateInduction).toHaveBeenCalledWith(prisonNumber, updateInductionDto, 'some-token')
+      expect(res.redirect).toHaveBeenCalledWith(`/plan/${prisonNumber}/view/work-and-interests`)
+      expect(req.session.skillsForm).toBeUndefined()
+      expect(req.session.inductionDto).toBeUndefined()
+    })
+
+    it('should update Induction given form is submitted where the only change is a removal of the work type OTHER', async () => {
```

Comment:
Just wondering why you're differentiating between removing OTHER and non OTHER? I could understand editing OTHER, but not removing?

Reply:
That is a good point; I think it was from my original thought process (which we spoke about on Friday) .... I'll have a quick think about renaming it (or removing it completely) .... 🤔

Reply:
Removed the test 👍